File: app/src/test/java/com/swensun/potato/List_Test.java
package com.swensun.potato;

import org.junit.Test;

import java.util.ArrayList;

public class List_Test {

    @Test
    public void remove_test() {
        ArrayList<Integer> list = new ArrayList<Integer>();
        for (int i = 0; i < 50; i++) {
            list.add(i);
        }

        for (int ii: list) {
            if (ii == 10) {
                list.remove(ii);
            }
        }
    }
}

File: app/src/test/java/com/swensun/potato/List_Test.kt
package com.swensun.potato

import org.junit.Test

class List_Test {

    @Test
    fun add_test() {
        var l = arrayListOf<RInt>()
        l.addAll((0 until 10).map {
            RInt(it)
        })
        l.addAll((5 until 15).map { RInt(it) })
        println(l.distinctBy { it.id }.map { "${it.id}--${it.count}" })
    }
}

class RInt(var id: Int) {
    var count = id
}

